src/notte/common/credential_vault/hashicorp/vault.py

```diff
@@ -1,7 +1,9 @@
+import os
 from dataclasses import dataclass
 from typing import Any, Protocol, final
 from urllib.parse import urlparse
 
+from dotenv import load_dotenv
```

Comment:
That should not be there. This line should be moved back to `vault_agents.py`. It's the user's responsability to load their env file (in whatever way they want to).